x-pack/plugins/apm/public/components/shared/MetadataTable/SpanMetadata/index.tsx

```diff
@@ -6,19 +6,41 @@
  */
 
 import React, { useMemo } from 'react';
-import { SPAN_METADATA_SECTIONS } from './sections';
 import { Span } from '../../../../../typings/es_schemas/ui/span';
-import { getSectionsWithRows } from '../helper';
+import { getSectionsFromFields } from '../helper';
 import { MetadataTable } from '..';
+import { FETCH_STATUS, useFetcher } from '../../../../hooks/use_fetcher';
+import { ProcessorEvent } from '../../../../../common/processor_event';
 
 interface Props {
   span: Span;
 }
 
 export function SpanMetadata({ span }: Props) {
-  const sectionsWithRows = useMemo(
-    () => getSectionsWithRows(SPAN_METADATA_SECTIONS, span),
-    [span]
+  const { data: spanEvent, status } = useFetcher(
+    (callApmApi) => {
+      return callApmApi({
+        endpoint: 'GET /api/apm/event_metadata/{processorEvent}/{id}',
+        params: {
+          path: {
+            processorEvent: ProcessorEvent.span,
+            id: span.span.id,
+          },
```

Comment:
I think it would be more straightforward to produce the filter on the client:

```suggestion
          query: {
            kuery: `span.id: ${span.span.id}`
          },
```

Right now we construct the filter on the server based on the `processor.event`.